sudoku-django-project/mainapp/classes.py

```diff
@@ -0,0 +1,284 @@
+
+        # Recursive method to check whether there is a unique solution for a number removed from grid
+        for i in range(81):
+
+            row = i // 9
+            col = i % 9
+
+            if grid[row][col] == 0:
+                for value in range(10):
+                    if value in self.possible_cell_values(row, col):
+                        grid[row][col] = value
+                        self.set_value_of_grid(grid, "row")
+                        if self.check_complete(grid):
+                            self.counter += 1
+                            break
+                        else:
+                            if self.unique_sol_check(grid):
+                                return True
+                break
+        grid[row][col] = 0
+        self.set_value_of_grid(grid, "row")
+
+    def puzzle_maker(self) -> None:
+
+        # adds spaces to the finished board
+        attempts = 5
+        while attempts > 0:
+            row = randint(0, 8)
+            col = randint(0, 8)
+            while self[row][col].get_value() == 0:
+                row = randint(0, 8)
+                col = randint(0, 8)
+            backup = self[row][col].get_value()
+            self[row][col].set_value(0)
+
+            copy = self.get_all_row_values()
+
+            self.counter = 0
+            self.unique_sol_check(copy)
+
+            if self.counter != 1:
+                self[row][col].set_value(backup)
+                attempts -= 1
+
+    def check(self, user_input: list) -> bool or list:
+
+        if user_input == self.full_board:
+            return True
+        else:
+            return [[True if self.full_board[row][col] == user_input[row][col] else False for col in range(9)] for row in range(9)]
+
+    # operator overloading methods.
+    def __iter__(self):  # noqa: ANN204
+        for i in self.top_boxes:  # x here is a box
+            yield i  # iterates through the boxes in the same way as
+        for i in self.mid_boxes:  # a matrix; i.e. left to right.
+            yield i
+        for i in self.bottom_boxes:
+            yield i
+
+    def __getitem__(self, index: int):  # noqa: ANN204
+        count = 0  # returns box object at index 3
+        for i in self:
+            if count == index:
+                return i
+            count += 1
+
+    def __eq__(self, b: "Sudoku") -> bool:  # functionality -> sudoku1 == sudoku2
+        for box in range(9):  # compares all values of both sudokus.
+            for element in range(9):
+                if self[box][element].get_value() != b[box][element].get_value(): return False
+        return True
+
+
+class _Box:
+    """ A class that acts like one of the nine 3x3 boxes in sudoku. """
+
+    def __init__(self):  # noqa: ANN204
+        self.top_row = [_Element() for _ in range(3)]
+        self.mid_row = [_Element() for _ in range(3)]
+        self.bottom_row = [_Element() for _ in range(3)]
+        self.box_pos = None  # NOT USED
```

Comment:
Can remove the box pos, and the two methods that are being run here as well as their definitions, if they don't break anything in your code